javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java

```diff
@@ -150,12 +152,18 @@ private MethodUsage resolveMethodTypeParametersFromExplicitList(TypeSolver typeS
     @Override
     public SymbolReference<ResolvedMethodDeclaration> solveMethod(String name, List<ResolvedType> argumentsTypes, boolean staticOnly) {
         Collection<ResolvedReferenceTypeDeclaration> rrtds = findTypeDeclarations(wrappedNode.getScope());
+
+        if (rrtds.isEmpty()) {
```

Comment:
Question: Is there a reason that we're only adding it if it's empty? Could/should we do it in all cases?

There might be an obvious answer here (or in linked discussions) that I'm not seeing while on mobile!

Reply:
if we add it when it's not empty then we're effectively adding a bound that's almost certainly more permissive than the actual type. consider `<T extends InputStream>` - if we add `Object` unconditionally here then we'd effectively be transforming that type parameter into `<T extends InputStream | Object>`.